### DISCUSSION TITLE
GNSS data into LOS

### DISCUSSION THREAD

**Discussion by sunzhe-96** [gmtsar/gmtsar#343] [General]
Hello,everyone
I  construct a GNSS_vel.enu file, then I need to enter the velocities from each station in east, north, up components, where should I go to get the velocities of the enu, is it in the ven3 file? just like the txt?
[G100.txt](https://github.com/gmtsar/gmtsar/files/8899069/G100.txt)

**Comment by Xiaohua-Eric-Xu** [comment]
You could get velocities estimated. But this might be better from someone who does analysis on GNSS data. You could also integrate at interferogram level as in https://doi.org/10.1029/2021JB022579 and https://doi.org/10.1093/gji/ggac121 . 
@kguns32 any thoughts?

**Reply by sunzhe-96** [reply]
> You could get velocities estimated. But this might be better from someone who does analysis on GNSS data. You could also integrate at interferogram level as in https://doi.org/10.1029/2021JB022579 and https://doi.org/10.1093/gji/ggac121 . @kguns32 any thoughts?

So the GNSS data I downloaded from the http://geodesy.unr.edu/NGLStationPages/gpsnetmap/GPSNetMap.html without getting velocities estimated cannot be used to compare the sar results directly, right? I always thought that converting the downloaded GNSS elevation data to LOS can be compared.

**Reply by kguns32** [reply]
If you're just comparing InSAR velocities with GNSS velocities, then you need to grab the velocity for whatever stations you want, and convert those velocities to LOS. UNR has velocities at this page http://geodesy.unr.edu/index.php just scroll down on the right side of the screen under MIDAS velocities and pick the IGS14 reference frame file and search for your station. 

Are you trying to compare GNSS time series with an InSAR pixel's time series? If so, then you can use the GNSS time series file you have (G100.txt) and covert the ENU displacements to LOS and plot that time series, [but keep in mind that unless you integrate the InSAR interferogram(s) with GNSS as Xiaohua suggested, they will be in different reference frames.]

**Reply by sunzhe-96** [reply]
Thanks for your reply @kguns32 ! I want to compare GNSS time series with an InSAR pixel's time series, but I have no idea how to convert these time series data to LOS, although I looked at sentinel_time_series_2.pdf.

**Reply by kguns32** [reply]
Gotcha -- have you seen our more updated time series document (attached)? We have added more explanations of how to project GNSS time series into LOS, near the end of the document. There is a tool now in the GMTSAR distribution that should accomplish this for you too, called gnss_enu2los.csh. Check out its usage information, but you should be able to give it ENU displacements and it will return LOS as one of its outputs. 

[Refined_TimeSeries_Documentation_June09_2022.pdf](https://github.com/gmtsar/gmtsar/files/8910567/Refined_TimeSeries_Documentation_June09_2022.pdf)

**Reply by sunzhe-96** [reply]
> Gotcha -- have you seen our more updated time series document (attached)? We have added more explanations of how to project GNSS time series into LOS, near the end of the document. There is a tool now in the GMTSAR distribution that should accomplish this for you too, called gnss_enu2los.csh. Check out its usage information, but you should be able to give it ENU displacements and it will return LOS as one of its outputs.
> 
> [Refined_TimeSeries_Documentation_June09_2022.pdf](https://github.com/gmtsar/gmtsar/files/8910567/Refined_TimeSeries_Documentation_June09_2022.pdf)

Thanks for your help kguns32! I will try to read the updated time series document!

**Reply by Congzhangg** [reply]
> 如果您只是将InSAR速度与GNSS速度进行比较，那么您需要获取所需的任何站点的速度，并将这些速度转换为LOS。UNR在此页面上有速度[，http://geodesy.unr.edu/index.php](http://geodesy.unr.edu/index.php) 只需在屏幕右侧的MIDAS速度下向下滚动，然后选择IGS14参考系文件并搜索您的电台。
> 
> 是否要将 GNSS 时间序列与 InSAR 像素的时间序列进行比较？如果是这样，那么你可以使用你拥有的GNSS时间序列文件（G100.txt）并将ENU位移隐蔽到LOS并绘制该时间序列，[但请记住，除非你像小华建议的那样将InSAR干涉图与GNSS集成，否则它们将位于不同的参考系中。

Hello, could you please tell me how to get the speed of the research period of the site from the UNR interface?

**Reply by kguns32** [reply]
Do you mean the velocity? This page has velocities for every station, along with the time period over which that velocity was estimated: http://geodesy.unr.edu/velocities/midas.IGS14.txt  the readme to understand every column on that page is here: http://geodesy.unr.edu/velocities/midas.readme.txt

If you want individual displacement time series, you need to go to the individual station page (full list here: http://geodesy.unr.edu/NGLStationPages/GlobalStationList )

**Reply by Congzhangg** [reply]
![1666145705105](https://user-images.githubusercontent.com/112800803/196581458-fa564bd3-d47f-45e8-8fed-2be29ada6a74.png)
I know this。 please tell me how do I calculate the speed of my research time

**Reply by kguns32** [reply]
I am not sure what you mean when you say "speed of my research time"? What is your research time? Do you mean velocity of the station? If you have a different time period over which you want to estimate a velocity (i.e., not 2009.0048 - 2022.7652, but some other time window), you will need to download the time series and estimate it yourself. Since that can be a challenge, you can make an assumption that the velocity estimated over this time window is similar to any other time window, but that is an assumption.

**Reply by Congzhangg** [reply]
I understand what you mean. My research period is from August 2017 to August 2020. Because it is challenging to calculate the velocity of the study time based on the temporal displacement series, it can be assumed that the rate of change is the same for each year。right?

**Reply by kguns32** [reply]
Gotcha! Yes, you can. It is an assumption that should be noted though (you're essentially just trusting that UNR's velocity model has a good fit to the time series, which for some stations may not necessarily be true). You could think of it like an approximate velocity.

**Reply by Congzhangg** [reply]
Thank you very much for your answer and have a nice day! (smile~)

**Reply by Congzhangg** [reply]
> Gotcha! Yes, you can. It is an assumption that should be noted though (you're essentially just trusting that UNR's velocity model has a good fit to the time series, which for some stations may not necessarily be true). You could think of it like an approximate velocity.

Hello，I am trying to correct 203 interference pairs with GPS data. Do you have available cyclic csh files？

**Reply by Xiaohua-Eric-Xu** [reply]
Usually cshell could get a for loop with, e.g.

```
foreach variable (`cat list`)
  # do your stuff here
end
```
where list contains all your variable
